libsession/src/main/java/org/session/libsession/messaging/messages/control/MessageRequestResponse.kt

```diff
@@ -26,7 +33,13 @@ class MessageRequestResponse(val isApproved: Boolean) : ControlMessage() {
         fun fromProto(proto: SignalServiceProtos.Content): MessageRequestResponse? {
             val messageRequestResponseProto = if (proto.hasMessageRequestResponse()) proto.messageRequestResponse else return null
             val isApproved = messageRequestResponseProto.isApproved
-            return MessageRequestResponse(isApproved)
+            val profileProto = messageRequestResponseProto.profile
+            val profile = Profile().apply {
+                displayName = profileProto.displayName
+                profileKey = messageRequestResponseProto.profileKey.toByteArray()
```

Comment:
the only possible issues I could see is from this assuming `profileKey` or `profilePicture` are non-null, if the user doesn't have a profile picture (just the letters in avatar) then it these fields will be null which could be used in the Profile object that's used later, but assuming they are non null here is probably fine from a runtime perspective, just that we might accidentally be storing them as empty strings in the case of profilePictureUrl or empty array in the case of the profileKey, which might be treated differently to null somewhere else in the application

Reply:
Good catch, I'll update.